File: 1. Two Sum/solution.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $stored = [];

        foreach ($nums as $key => $num){

            if(array_key_exists($num, $stored)){
                return [$stored[$num], $key];
            }

            $required = $target - $num;

            $stored[$required] = $key;
        }
    }
}

File: 1. Two Sum/solution.go
func twoSum(nums []int, target int) []int {
    stored := make(map[int]int, len(nums))

    for index, num := range nums {
        if required, ok := stored[num]; ok {
            return []int{required, index}
        }

        stored[target - num] = index
    }

    return nil
}
